mlflow/genai/evaluation/base.py

```diff
@@ -279,6 +279,14 @@ def predict_fn(question: str) -> str:
             message=r"Hint: Inferred schema contains integer column\(s\).*",
             category=UserWarning,
         )
+        # Suppress numpy warning about ragged nested sequences. This is raised when passing
+        # a dataset that contains complex object to mlflow.evaluate(). MLflow convert data
+        # into numpy array to compute dataset digest, which triggers the warning.
+        warnings.filterwarnings(
+            "ignore",
+            message=r"Creating an ndarray from ragged nested sequences",
+            module="mlflow.data.evaluation_dataset",
+        )
 
         result = mlflow.models.evaluate(
```

Comment:
```suggestion
        with warnings.catch_warnings():
            warnings.filterwarnings(
            "ignore",
            message=r"Creating an ndarray from ragged nested sequences",
            module="mlflow.data.evaluation_dataset",
        )

            result = mlflow.models.evaluate(
```

can we use `catch_warnings`?

Reply:
@harupy we have `catch_warnings` in L276.

Reply:
oh I missed it :)